experiment_prototype/scan_classes/sequences.py

```diff
@@ -104,203 +82,143 @@ def __init__(self, seqn_keys, sequence_slice_dict, sequence_interface, transmit_
-        :param sequence_interface: the interfacing dictionary that describes how to interface the 
+        :param sequence_interface: the interfacing dictionary that describes how to interface the
          slices that are included in this sequence. Keys are tuples of format
-         (slice_id_1, slice_id_2) and values are of interface_types set up in 
+         (slice_id_1, slice_id_2) and values are of interface_types set up in
          experiment_prototype.
         :param transmit_metadata: metadata from the config file that is useful here.
         """
 
-        # TODO make diagram(s) for pulse combining algorithm
-        # TODO make diagram for pulses that are repeats, showing clearly what intra_pulse_start_time,
-        # and pulse_shift are.
         ScanClassBase.__init__(self, seqn_keys, sequence_slice_dict, sequence_interface,
                                transmit_metadata)
 
-        # TODO: pass clear frequencies to pass to pulses
 
-        # All interfacing at this point is PULSE.
+        txrate = self.transmit_metadata['txrate']
+        txctrfreq = self.transmit_metadata['txctrfreq']
+        main_antenna_count = self.transmit_metadata['main_antenna_count']
+        main_antenna_spacing = self.transmit_metadata['main_antenna_spacing']
+        pulse_ramp_time = self.transmit_metadata['pulse_ramp_time']
+        max_usrp_dac_amplitude = self.transmit_metadata['max_usrp_dac_amplitude']
+        tr_window_time = self.transmit_metadata['tr_window_time']
+
+        self.basic_slice_pulses = {}
+        single_pulse_timing = []
+        for slice_id in self.slice_ids:
+
+            exp_slice = self.slice_dict[slice_id]
+            tx_freq_khz = float(exp_slice['txfreq'])
+            wave_freq = tx_freq_khz - txctrfreq
+            wave_freq_hz = wave_freq * 1000
+
+            if not exp_slice['rxonly']:
+                basic_samples, real_freq = get_samples(txrate, wave_freq_hz,
+                                                   float(exp_slice['pulse_len']) / 1e6,
+                                                   pulse_ramp_time,
+                                                   1.0,
+                                                   exp_slice['iwavetable'],
+                                                   exp_slice['qwavetable'])
+                if real_freq != wave_freq_hz:
+                    errmsg = 'Actual Frequency {} is Not Equal to Intended Wave Freq {}'.format(real_freq,
+                                                                                                wave_freq_hz)
+                    raise ExperimentException(errmsg)  # TODO change to warning? only happens on non-SINE
+
+                # convert the beam angles to rads
+                beam_rads = np.pi / 180 * np.array(exp_slice['beam_angle'], dtype=np.float64)
+
+                main_antennas = np.arange(main_antenna_count)
+                x = (((main_antenna_count - 1) / 2.0 - main_antennas) * main_antenna_spacing)
```

Comment:
ok, so I hear you that you have updated this and want to just do RX later, but I really think that beamforming code needs to be the same lines of code running for RX and TX. I think these lines need to replace get_phshift in a function and then rx_azimuth_to_antenna_offset can use the same function (not too big of a code change). Diverging the code used for RX and TX is a detriment to code quality imo (increased chance for bugs in future if we forget they are separate, with direct effect on data produced)

Reply:
hey @kkotyk, in the other comment you said: 
> I've mainly focused on tx with the PR, but I think once TX is verified, I'll work on bundling RX code into this as well as debug samples.

Are you waiting on me for TX verification (do you mean looking at the loopback data)?